docs(aws): aws kms authentication procedures

PaulRMellor opened the pull request:
Documentation

Refactor of the AWS KMS authentication sections to create clear procedures/steps for the two options

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Update documentation
- [ ] Make sure all unit/integration tests pass
- [ ] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [ ] If applicable to the change, [trigger the system test suite](../blob/main/DEV_GUIDE.md#jenkins-pipeline-for-system-tests).  Make sure tests pass.
- [ ] If applicable to the change, [trigger the performance test suite](../blob/main/PERFORMANCE.md#jenkins-pipeline-for-performance). Ensure that any degradations to performance numbers are understood and justified.
- [ ] Ensure the PR references relevant issue(s) so they are closed on merging.
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

> **_NOTE:_**  You must be a member of `@kroxylicious/developers` to trigger the system test and performance test suites.  If you are not part of this group, comment on the PR requesting a trigger, tagging `@kroxylicious/developers`.

k-wall commented on docs/modules/record-encryption/aws-kms/con-aws-kms-service-config-identity-ec2-metadata.adoc:
```diff
@@ -3,23 +3,24 @@
-    Defaults to `0.8`, Thus means the credential will refreshed once it reaches 80% of its lifetime.
-<6> AWS region identifier (e.g. `us-east-1`).
+<4> (Optional) Metadata endpoint used to obtain EC2 metadata. 
+Defaults to `\http://169.254.169.254/`.
```
Is the leading `\` a typo?

PaulRMellor replied:
It's to escape the formatting on the example, so it doesn't become a link in the output.

k-wall replied:
Ah, I didn't know that.

k-wall commented on docs/modules/record-encryption/aws-kms/con-aws-kms-service-config.adoc:
```diff
@@ -5,7 +5,8 @@
 
-For AWS KMS, the KMS configuration looks like this.
+This section provides examples of AWS KMS configurations for authenticating with AWS KMS services. 
+Use these examples to configure Kroxylicious to authenticate using either a long-term IAM identity or EC2 instance metadata credentials.
```
How will this be handled downstream?

PaulRMellor replied:
Yes. I've reduced to fit up/down

PaulRMellor commented:
> This section provides examples of

Thanks @k-wall . I've addressed the comments.  The leading `/` escapes the formatting so we don't get a link in the output. I've changed the lead sentence for the examples so it fits downstream.

k-wall reviewed (approved):
lgtm
